src/System.Runtime/tests/System/LazyOfTMetadata.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using Xunit;
+
+namespace System.Runtime.Tests
+{
+    public static class LazyOfTMetadataTests
```

Comment:
This looks good to me. I actually added a few test cases for this type for .NET Native when I made a bug fix for it there. Yours seems to cover everything mine did and more, although I have a test case that checks a factory method which throws (i.e. new Lazy<int>( () => throw new Exception() ... ), which I don't see here.